soroban-sdk/Cargo.toml

```diff
@@ -36,7 +36,7 @@ ctor = { version = "0.2.1", optional = true }
 
 [dev-dependencies]
 soroban-sdk-macros = { workspace = true, features = ["testutils"] }
-soroban-env-host = { workspace = true, features = ["testutils"] }
+soroban-env-host = { workspace = true, features = ["testutils", "unstable-next-api"] }
```

Comment:
The idea behind the feature was to get aroudn semver checks in controlled fashion for the repos which we own (such as soroban-rpc), as tying the env version to protocol version doesn't work well for all the cases (like this one).
SDK update is not critical, but it would be nicer to get changes gradually. I'm not sure why should anyone import both host and SDK at the same time, but if you're concerned we could wait as well.

Reply:
Actually, soroban-rpc does use both env and SDK (though not for the same crates, which I was referring to later). Consistency checks in Soroban RPC want SDK to depend on the same version of env, as the soroban RPC dependency version. I'm actually if this check is necessary anymore, but in order to work around it we either need to have this PR merged, or we need to disable it.

Reply:
Also, I'm not confident if the feature is actually forced on the SDK users, IIUC features are not transitive, are they?

Reply:
i.e. It's best if the binary / final target enables the unstable api, rather than libs.

Reply:
FWIW we don't need to enable this feature in exported RPC libraries, only in the internal preflight crate (which is technically not a binary itself, but is not published and is only linked to the go binary).

Reply:
Makes sense. We just need to remember that there’s a small amount of api adoption work to do later. 👍